src/master/init.h

```diff
@@ -35,13 +38,27 @@
 #include "master/matomlserv.h"
 #include "master/personality.h"
 #include "master/topology.h"
+#include "metrics/metrics.h"
 
 /// Functions to call before normal startup
 inline const std::vector<RunTab> earlyRunTabs = {
     RunTab{metadataserver::personality_validate, "validate personality"}};
 
 /// Functions to call during normal startup
 inline const std::vector<RunTab> runTabs = {
+    RunTab{[]() {
+	           if (cfg_getuint8("ENABLE_PROMETHEUS", 0) != 1) {
+		           safs::log_info(
+		               "Prometheus disabled, no Prometheus metrics will be "
+		               "gathered");
+		           return 0;
+	           }
+	           metrics::metrics_init(
+	               cfg_getstr("PROMETHEUS_HOST", "0.0.0.0:8080"));
+	           eventloop_destructregister(metrics::metrics_destroy);
+	           return 0;
```

Comment:
Consider splitting the lambda function subsection into smaller function, for code readability. If you consider it useful try something like this:  
```cpp
// Define a function for initializing Prometheus metrics
int init_prometheus_metrics() {
    auto prometheus_host = cfg_getstr("PROMETHEUS_HOST", "0.0.0.0:8080");
    metrics::metrics_init(prometheus_host);
    eventloop_destructregister(metrics::metrics_destroy);
    return 0;
}

// Update the lambda function to use the new helper function
inline const std::vector<RunTab> runTabs = {
    RunTab{[]() -> int {
        if (cfg_getuint8("ENABLE_PROMETHEUS", 0)!= 1) {
            safs::log_info("Prometheus disabled, no Prometheus metrics will be gathered");
            return 0;
        }
        return init_prometheus_metrics();
    }, "prometheus module"}
};```

Reply:
Mhm, good suggestion, but I would probably go even further and make it entirely it's own function that is then part of runTabs. I'll think about it and tinker a bit.